core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -53,6 +55,7 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
         this.clock = clock;
         this.channelMonitorAgent = channelMonitorAgent;
         this.validator = validator;
+        this.collectorManager = new CollectorManager(topic, channelManager);
```

Comment:
Consumer is used per topic,and holds collections of Collectors(inside CollectorManager) which are listening this topic.